crates/libcontainer/src/process/container_main_process.rs

```diff
@@ -107,15 +108,22 @@ pub fn container_main_process(container_args: &ContainerArgs) -> Result<Pid> {
     // process is exit and reaped. By this point, the intermediate process
     // should already exited successfully. If intermediate process errors out,
     // the `init_ready` will not be sent.
-    match waitpid(intermediate_pid, None)? {
-        WaitStatus::Exited(_, 0) => (),
-        WaitStatus::Exited(_, s) => {
+    match waitpid(intermediate_pid, None) {
+        Ok(WaitStatus::Exited(_, 0)) => (),
+        Ok(WaitStatus::Exited(_, s)) => {
             log::warn!("intermediate process failed with exit status: {s}");
         }
-        WaitStatus::Signaled(_, sig, _) => {
+        Ok(WaitStatus::Signaled(_, sig, _)) => {
             log::warn!("intermediate process killed with signal: {sig}")
         }
-        _ => (),
+        Ok(_) => (),
+        Err(err) => {
+            if err == Errno::ECHILD {
```

Comment:
Nit: Can this be turned into `Err(Errno::ECHILD)` case?

Reply:
I think we don't need to do it because `waitpid` returns  [Result](https://doc.rust-lang.org/nightly/core/result/enum.Result.html)<T, [Errno](https://docs.rs/nix/latest/nix/errno/enum.Errno.html)>
https://docs.rs/nix/latest/nix/type.Result.html